pkg/client/kubernetes/runtime_client.go

```diff
@@ -98,6 +101,37 @@ func setClientOptionsDefaults(config *rest.Config, options *client.Options) erro
 	return nil
 }
 
+// AggregatorCacheFunc returns a `cache.NewCacheFunc` which creates a cache that holds different cache implementations depending on the objects' GVKs.
+func AggregatorCacheFunc(newCache cache.NewCacheFunc, typeToNewCache map[client.Object]cache.NewCacheFunc, scheme *runtime.Scheme) cache.NewCacheFunc {
+	return func(config *rest.Config, options cache.Options) (cache.Cache, error) {
+		if err := setCacheOptionsDefaults(&options); err != nil {
+			return nil, err
+		}
+
+		fallbackCache, err := newCache(config, options)
+		if err != nil {
+			return nil, err
+		}
+
+		gvkToCache := make(map[schema.GroupVersionKind]cache.Cache)
+		for object, fn := range typeToNewCache {
+			gvks, _, err := scheme.ObjectKinds(object)
+			if err != nil || len(gvks) != 1 {
```

Comment:
Nit: if `len(gvks) > 1` `err` is probably `nil`, so we're returning `nil` here in that case.
Can you use `apiutil.GVKForObject` (which handles this case already) instead of `scheme.ObjectKinds`?